nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/ConsumerPool.java

```diff
@@ -136,37 +134,34 @@ public ConsumerPool(
-                    // Create reader group manager.
-                    readerGroupManager = ReaderGroupManager.withScope(scope, clientConfig);
-                    try {
-                        if (haveReaderGroup) {
-                            // Use the reader group from the previous state.
-                            logger.debug("ConsumerPool: previous state={}", new Object[]{stateMap.toMap()});
-                            final String prevReaderGroupName = stateMap.get(STATE_KEY_READER_GROUP);
-                            if (prevReaderGroupName == null || prevReaderGroupName.isEmpty()) {
-                                throw new Exception("State is missing the reader group name.");
-                            }
-                            readerGroupName = prevReaderGroupName;
-                            logger.debug("ConsumerPool: Using existing reader group {}", new Object[]{readerGroupName});
-                            readerGroup = readerGroupManager.getReaderGroup(readerGroupName);
-                        } else {
-                            if (!primaryNode) {
-                                throw new RuntimeException("bug");
-                            }
-                            // This processor is starting for the first time on the primary node.
+                    if (haveReaderGroup) {
+                        // Use the reader group from the previous state.
+                        logger.debug("ConsumerPool: previous state={}", new Object[]{stateMap.toMap()});
+                        final String prevReaderGroupName = stateMap.get(STATE_KEY_READER_GROUP);
+                        if (prevReaderGroupName == null || prevReaderGroupName.isEmpty()) {
+                            throw new Exception("State is missing the reader group name.");
+                        }
+                        readerGroupName = prevReaderGroupName;
+                        logger.debug("ConsumerPool: Using existing reader group {}", new Object[]{readerGroupName});
+                        readerGroup = readerGroupManager.getReaderGroup(readerGroupName);
+                    } else {
+                        if (!primaryNode) {
+                            throw new RuntimeException("bug");
+                        }
+                        // This processor is starting for the first time on the primary node.
 
-                            // Create streams.
-                            try (final StreamManager streamManager = StreamManager.create(clientConfig)) {
-                                for (Stream stream : streams) {
-                                    streamManager.createScope(stream.getScope());
-                                    streamManager.createStream(stream.getScope(), stream.getStreamName(), streamConfig);
-                                }
+                        // Create streams.
+                        try (final StreamManager streamManager = StreamManager.create(clientConfig)) {
```

Comment:
`streamManager` needed to have a longer lifetime which enabled the removal of `batchClient`.